Merge with Develop

jaskaransinghsidana opened the pull request:
This changes creates the main branch from the develop with all the changes as on 2nd May

jaskaransinghsidana committed:
Merge pull request #6 from NeerajMalhotra-QB/feature_pyspark_backend

Feature pyspark backend

NeerajMalhotra-QB committed:
add pyspark model components and types

NeerajMalhotra-QB committed:
remove hypothesis

NeerajMalhotra-QB committed:
remove synthesis and hypothesis

jaskaransinghsidana committed:
Pyspark backend components, base, container, accessor, test file for accessor

jaskaransinghsidana committed:
Merge pull request #7 from NeerajMalhotra-QB/feature_pyspark_backend

Pyspark backend components changes

NeerajMalhotra-QB committed:
pushing changes to model components, engine

NeerajMalhotra-QB committed:
Merge pull request #8 from NeerajMalhotra-QB/model_comps

add pyspark changes to model components, engine, etc

NeerajMalhotra-QB committed:
test for pyspark dataframeschema class

NeerajMalhotra-QB committed:
test schema with alias types

NeerajMalhotra-QB committed:
ensuring treat dataframes as tables types

NeerajMalhotra-QB committed:
update container for pyspark dataframe

NeerajMalhotra-QB committed:
adding negative test flow

NeerajMalhotra-QB committed:
removing series and index on pysparrk dataframes

NeerajMalhotra-QB committed:
remove series

NeerajMalhotra-QB committed:
revert series from pyspark.pandas

NeerajMalhotra-QB committed:
adding checks for pyspark

NeerajMalhotra-QB committed:
registering pysparkCheckBackend

NeerajMalhotra-QB committed:
cleaning base

NeerajMalhotra-QB committed:
Merge pull request #9 from NeerajMalhotra-QB/pyspark_schema

Pyspark schema

jaskaransinghsidana committed:
Fixing the broken type cast check, validation of schema fix.

jaskaransinghsidana committed:
Merge pull request #10 from NeerajMalhotra-QB/feature_pyspark_backend

Fixing the typecast and Validate function

NeerajMalhotra-QB committed:
define spark level schema

NeerajMalhotra-QB committed:
fixing check flow

NeerajMalhotra-QB committed:
setting apply fn

NeerajMalhotra-QB committed:
add sub sample functionality

NeerajMalhotra-QB committed:
adjusting test case against common attributes

NeerajMalhotra-QB committed:
need apply for column level check

NeerajMalhotra-QB committed:
adding builtin checks for pyspark

NeerajMalhotra-QB committed:
adding checks for pyspark df

NeerajMalhotra-QB committed:
getting check registered

NeerajMalhotra-QB committed:
fixing a bug a in error handling for schema check

NeerajMalhotra-QB committed:
check_name validation fixed

NeerajMalhotra-QB committed:
implementing dtype checks for pyspark

NeerajMalhotra-QB committed:
updating error msg

NeerajMalhotra-QB committed:
fixing dtype reason_code

NeerajMalhotra-QB committed:
updating builtin checks for pyspark

NeerajMalhotra-QB committed:
registeration

NeerajMalhotra-QB committed:
Merge pull request #11 from NeerajMalhotra-QB/pyspark_schema

support pyspark checks, validations, schema matching, etc

jaskaransinghsidana committed:
Implementation of checks import and spark columns information check

jaskaransinghsidana committed:
Merge pull request #12 from NeerajMalhotra-QB/feature_pyspark_backend

Implementation of checks import and spark columns information check

NeerajMalhotra-QB committed:
enhancing __call__, checks classes and builtin_checks

NeerajMalhotra-QB committed:
Merge pull request #13 from NeerajMalhotra-QB/pyspark_builtin_checks

enhancing __call__, checks classes and builtin_checks

I don’t think we need to add another arg column_name

 backend = self.get_backend(check_obj)(self)
        return backend(check_obj, column, column_name)
since column is already present there.

Moreover, this is in pandera/api/checks.py which will impact pandas and others functionality too.. So I reverted this code back for now…

There’s a kwargs which we can leverage as follows:

@register_builtin_check(
    error="str_startswith('{string}')",
)
def str_startswith(data: DataFrame, string: str, kwargs: dict) -> bool:
    """Ensure that all values start with a certain string.

    :param string: String all values should start with
    :param kwargs: key-word arguments passed into the `Check` initializer.
    """
    breakpoint()  # TODO: change to accept column and perform check on it

    return True 
with this I can access both column and value to check

> /Users/Neeraj_Malhotra/qb_assets/os/forked/pandera/pandera/backends/pyspark/builtin_checks.py(226)str_startswith()
-> breakpoint()  # TODO: change to accept column and perform check on it
(Pdb) kwargs
{'string': 'B'}
(Pdb) string
'product'
(Pdb) data.show()
+-------+-----+
|product|price|
+-------+-----+
|  Bread|    9|
| Butter|   15|
+-------+-----+
Next we can do something like return df.filter(condition).limit(1) == 1 as boolean value as check result.

NeerajMalhotra-QB committed:
delete junk files

NeerajMalhotra-QB committed:
Merge branch 'pyspark_builtin_checks' into develop

jaskaransinghsidana committed:
Changes to fix the implemtation of checks. Changed Apply function to send list with dataframe and column name, builtin function registers functions with lists which inculdes the dataframe

jaskaransinghsidana committed:
Merge pull request #14 from NeerajMalhotra-QB/feature_pyspark_backend

Changes to fix the implemtation of checks. Changed Apply function to …

NeerajMalhotra-QB committed:
extending pyspark checks

NeerajMalhotra-QB committed:
Merge pull request #15 from NeerajMalhotra-QB/ps_err

extending pyspark checks

jaskaransinghsidana committed:
Fixed builtin check bug and added test for supported builtin checks for pyspark

jaskaransinghsidana committed:
Merge pull request #16 from NeerajMalhotra-QB/feature_pyspark_backend

Fixed builtin check bug and added test for supported builtin checks f…

NeerajMalhotra-QB committed:
add todos

NeerajMalhotra-QB committed:
bydefault validate all checks

NeerajMalhotra-QB committed:
Merge pull request #17 from NeerajMalhotra-QB/ps_lazy_checks

bydefault validate all checks

NeerajMalhotra-QB committed:
fixing issue with sqlctx

NeerajMalhotra-QB committed:
Merge pull request #18 from NeerajMalhotra-QB/ps_fix_err_func

fixing issue with sqlctx

NeerajMalhotra-QB committed:
add dtypes pytests

NeerajMalhotra-QB committed:
setting up schema

NeerajMalhotra-QB committed:
add negative and positive tests

NeerajMalhotra-QB committed:
add fixtures and refactor tests

NeerajMalhotra-QB committed:
generalize spark_df func

NeerajMalhotra-QB committed:
refactor to use conftest

NeerajMalhotra-QB committed:
use conftest

NeerajMalhotra-QB committed:
Merge pull request #19 from NeerajMalhotra-QB/ps_dtyp_checks

adding tests for different dtypes supported in pyspark

NeerajMalhotra-QB committed:
Merge branch 'develop' into ps_check_ref

NeerajMalhotra-QB committed:
Merge pull request #20 from NeerajMalhotra-QB/ps_check_ref

refactoring pyspark checks

NeerajMalhotra-QB committed:
add support for decimal dtype and fixing other types

NeerajMalhotra-QB committed:
Merge pull request #21 from NeerajMalhotra-QB/ps_dtypes

add support for decimal dtype and fixing other types

jaskaransinghsidana committed:
Added new Datatypes support for pyspark, test cases for dtypes pyspark, created test file for error

jaskaransinghsidana committed:
Merge pull request #22 from NeerajMalhotra-QB/feature_pyspark_backend

Added new Datatypes support for pyspark, test cases for dtypes pyspar…

NeerajMalhotra-QB committed:
Merge branch 'develop' into ps_err_summarization

NeerajMalhotra-QB committed:
refactor ArraySchema

NeerajMalhotra-QB committed:
Merge pull request #23 from NeerajMalhotra-QB/column_cls

refactor ArraySchema

NeerajMalhotra-QB committed:
Merge branch 'develop' into ps_err_summarization

NeerajMalhotra-QB committed:
rename array to column.py

NeerajMalhotra-QB committed:
Merge pull request #24 from NeerajMalhotra-QB/ps_array

rename array to column

NeerajMalhotra-QB committed:
Merge branch 'develop' into ps_err_summarization

jaskaransinghsidana committed:
1) Changes in test cases to look for summarised error raise instead of fast fail, since default behaviour is changed to summarised.
2) Added functionality to accept and check the precision and scale in Decimal Datatypes.

jaskaransinghsidana committed:
Merge pull request #25 from NeerajMalhotra-QB/feature_pyspark_backend

Decimal Datatype functionality and summarised error raise fix

NeerajMalhotra-QB committed:
add neg test

NeerajMalhotra-QB committed:
add custom ErrorHandler

jaskaransinghsidana committed:
Added functionality to DayTimeIntervalType datatype to accept parameters

jaskaransinghsidana committed:
Added functionality to DayTimeIntervalType datatype to accept parameters

NeerajMalhotra-QB committed:
return summarized error report

NeerajMalhotra-QB committed:
replace dataframe to dict for return obj

jaskaransinghsidana committed:
Changed checks input datatype to custom named tuple from the existing list. Also started changing the pyspark checks to include more datatypes

jaskaransinghsidana committed:
Merge remote-tracking branch 'origin/feature_pyspark_backend' into feature_pyspark_backend

NeerajMalhotra-QB committed:
refactor

jaskaransinghsidana committed:
Merge pull request #26 from NeerajMalhotra-QB/feature_pyspark_backend

Feature pyspark backend

NeerajMalhotra-QB committed:
introduce error categories

NeerajMalhotra-QB committed:
rename error categories

NeerajMalhotra-QB committed:
fixing bug in schema.dtype.check

NeerajMalhotra-QB committed:
fixing error category to by dynamic

jaskaransinghsidana committed:
Added checks for each datatype in test cases. Reduced the code redundancy of the code in test file. Refactored the name of custom datatype object for checks.

NeerajMalhotra-QB committed:
error_handler pass through

jaskaransinghsidana committed:
Merge pull request #27 from NeerajMalhotra-QB/feature_pyspark_checks

Backend Checks

NeerajMalhotra-QB committed:
add ErrorHandler to column api

NeerajMalhotra-QB committed:
removed SchemaErrors since we now aggregate in errorHandler

NeerajMalhotra-QB committed:
fixing dict keys

jaskaransinghsidana committed:
Added Decorator to raise TypeError in case of unexpected input type for the check function.

jaskaransinghsidana committed:
Merge pull request #28 from NeerajMalhotra-QB/feature_pyspark_checks

Added Decorator to raise TypeError in case of unexpected input type f…

NeerajMalhotra-QB committed:
Merge branch 'develop' into ps_err_dict

NeerajMalhotra-QB committed:
Merge pull request #29 from NeerajMalhotra-QB/ps_err_dict

implementation of custom pyspark error handler

NeerajMalhotra-QB committed:
replace validator with report_errors

NeerajMalhotra-QB committed:
Merge pull request #30 from NeerajMalhotra-QB/new_validator

replace validator with report_errors

NeerajMalhotra-QB committed:
cleaning debugs

NeerajMalhotra-QB committed:
Merge pull request #31 from NeerajMalhotra-QB/remove_breaks

cleaning debugs

NeerajMalhotra-QB committed:
Support DataModels and Field

jaskaransinghsidana committed:
Added Decorator to raise TypeError in case of unexpected input type for the check function. Merged with Develop

jaskaransinghsidana committed:
Added Decorator to raise TypeError in case of unexpected input type for the check function. Merged with Develop

jaskaransinghsidana committed:
Fix to run using the class schema type

NeerajMalhotra-QB committed:
use alias types

NeerajMalhotra-QB committed:
clean up

NeerajMalhotra-QB committed:
Merge pull request #32 from NeerajMalhotra-QB/fields

add pyspark support for data models

NeerajMalhotra-QB committed:
add new typing for pyspark.sql

NeerajMalhotra-QB committed:
Merge pull request #33 from NeerajMalhotra-QB/pyspark-sql-typing

add new typing for pyspark.sql

jaskaransinghsidana committed:
Added changes to support raising error for use of datatype not supported by the check and support for map and array type.

jaskaransinghsidana committed:
Merge remote-tracking branch 'origin/feature_pyspark_checks' into feature_pyspark_checks

# Conflicts:
#	pandera/backends/pyspark/builtin_checks.py
#	tests/pyspark/test_pyspark_check.py

jaskaransinghsidana committed:
Merge pull request #34 from NeerajMalhotra-QB/feature_pyspark_checks

Feature pyspark checks

NeerajMalhotra-QB committed:
support bare dtypes for DataFrameModel

NeerajMalhotra-QB committed:
Merge branch 'develop' into bare_dtypes

NeerajMalhotra-QB committed:
Merge pull request #35 from NeerajMalhotra-QB/bare_dtypes

support bare dtypes (str, int, float, etc) for DataFrameModel flow

NeerajMalhotra-QB committed:
remove resolved TODOs and breakpoints

NeerajMalhotra-QB committed:
Merge pull request #36 from NeerajMalhotra-QB/cleanup

remove resolved TODOs and breakpoints

NeerajMalhotra-QB committed:
change to bare types

NeerajMalhotra-QB committed:
use spark types instead of bare types

NeerajMalhotra-QB committed:
using SchemaErrorReason instead of hardcode in container

NeerajMalhotra-QB committed:
fixing an issue with error reason codes

NeerajMalhotra-QB committed:
Merge pull request #37 from NeerajMalhotra-QB/error_reason

using SchemaErrorReason instead of hardcode in container

NeerajMalhotra-QB committed:
Merge branch 'develop' into field_tests

NeerajMalhotra-QB committed:
minor fix

NeerajMalhotra-QB committed:
Merge pull request #38 from NeerajMalhotra-QB/field_tests

support use of native pyspark types

NeerajMalhotra-QB committed:
fixing checks and errors in pyspark

NeerajMalhotra-QB committed:
Merge pull request #39 from NeerajMalhotra-QB/spark_checks

fixing checks and errors in pyspark

jaskaransinghsidana committed:
Changes include the following:
1) Updated dtypes test functionality to make it more readable
2) Changes in accessor tests to support the new functionality
3) Changes in engine class to conform to check class everywhere else

jaskaransinghsidana committed:
Merge pull request #40 from NeerajMalhotra-QB/feature_pyspark_checks

Changes to support Complex Types and Test case refactoring

NeerajMalhotra-QB committed:
enhancing dataframeschema and model classes

NeerajMalhotra-QB committed:
Merge branch 'develop' into update_columns

NeerajMalhotra-QB committed:
Merge pull request #41 from NeerajMalhotra-QB/update_columns

enhancing container, DataFrameSchema and model classes

jaskaransinghsidana committed:
Changes to remove the pandas dependency

jaskaransinghsidana committed:
Changes to remove the pandas dependency

jaskaransinghsidana committed:
Refactoring of the checks test functions

jaskaransinghsidana committed:
Refactoring of the checks test functions

jaskaransinghsidana committed:
Fixing the test case breaking

jaskaransinghsidana committed:
Merge pull request #42 from NeerajMalhotra-QB/fix_develop_test_failure

Fixing the test case breaking

jaskaransinghsidana committed:
Isort and Black formatting

jaskaransinghsidana committed:
Container Test function failure

jaskaransinghsidana committed:
Merge pull request #44 from NeerajMalhotra-QB/fix_develop_test_failure

Fix develop test failure

jaskaransinghsidana committed:
Isort and black linting

jaskaransinghsidana committed:
Isort and black linting

kasperjanehag reviewed (dismissed):
LGTM! Approved without review

jaskaransinghsidana committed:
Added Changes to refactor the checks class. Fixes to some test cases failures.

jaskaransinghsidana committed:
Merge remote-tracking branch 'origin/feature_pyspark_checks' into feature_pyspark_checks

# Conflicts:
#	pandera/__init__.py
#	pandera/accessors/pyspark_sql_accessor.py
#	pandera/api/pandas/types.py
#	pandera/backends/pyspark/base.py
#	pandera/backends/pyspark/builtin_checks.py
#	pandera/backends/pyspark/column.py
#	pandera/backends/pyspark/error_formatters.py
#	tests/pyspark/test_pyspark_check.py

jaskaransinghsidana committed:
Merge pull request #46 from NeerajMalhotra-QB/feature_pyspark_checks

Feature pyspark checks

jaskaransinghsidana committed:
Removing breakpoint

jaskaransinghsidana committed:
Merge pull request #47 from NeerajMalhotra-QB/feature_pyspark_checks

Removing breakpoint

NeerajMalhotra-QB committed:
fixing raise error

NeerajMalhotra-QB committed:
adding metadata dict

jaskaransinghsidana committed:
Removing the reference of pandas from docstrings

jaskaransinghsidana committed:
Merge pull request #48 from NeerajMalhotra-QB/fix_docstring_pyspark

Removing the reference of pandas from docstrings

jaskaransinghsidana committed:
Removing redundant code block in utils

jaskaransinghsidana committed:
Merge pull request #49 from NeerajMalhotra-QB/fix_docstring_pyspark

Removing redundant code block in utils

jaskaransinghsidana committed:
Changes to return dataframe with errors property

NeerajMalhotra-QB committed:
add accessor for errorHandler

NeerajMalhotra-QB committed:
support errors access on pyspark.sql

NeerajMalhotra-QB committed:
Merge pull request #50 from NeerajMalhotra-QB/err_accessor

support error accessor on pyspark dataframe

NeerajMalhotra-QB committed:
updating pyspark error tcs

NeerajMalhotra-QB committed:
fixing model test cases

NeerajMalhotra-QB committed:
adjusting errors to use pandera.errors

NeerajMalhotra-QB committed:
use accessor instead of dict

NeerajMalhotra-QB committed:
Merge pull request #51 from NeerajMalhotra-QB/align_tcs

align test cases to use pyspark.sql accessor

NeerajMalhotra-QB committed:
adding accessor

NeerajMalhotra-QB committed:
revert to  develop

NeerajMalhotra-QB committed:
Merge pull request #52 from NeerajMalhotra-QB/fix_docstring_pyspark

Fix docstring pyspark

jaskaransinghsidana committed:
Removal of imports which are not needed and improved test case.

jaskaransinghsidana committed:
Merge pull request #53 from NeerajMalhotra-QB/fix_docstring_pyspark

Removal of imports which are not needed and improved test case.

NeerajMalhotra-QB committed:
setting independent pyspark import

NeerajMalhotra-QB committed:
pyspark imports

NeerajMalhotra-QB committed:
Merge pull request #54 from NeerajMalhotra-QB/pyspark_imports

pyspark imports to be separate from pandas & pyspark.pandas to avoid conflicts

NeerajMalhotra-QB committed:
revert comments

NeerajMalhotra-QB committed:
Merge branch 'develop' into meta_tags

NeerajMalhotra-QB committed:
store and retrieve metadata at schema levels

NeerajMalhotra-QB committed:
adding metadata support

NeerajMalhotra-QB committed:
Merge pull request #55 from NeerajMalhotra-QB/meta_tags

support meta tags on column, field, model and schema levels

jaskaransinghsidana committed:
Added changes to support parameter based run.
1) Added parameters.yaml file to hold the configurations
2) Added code in utility to read the config
3) Updated the test cases to support the parameter based run
4) Moved pyspark decorators to a new file decorators.py in backend
5) Type fix in get_matadata property in container.py file

jaskaransinghsidana committed:
Changing the default value in config

NeerajMalhotra-QB committed:
Merge pull request #56 from NeerajMalhotra-QB/feature_kill_switch

Feature kill switch

NeerajMalhotra-QB committed:
change to consistent interface

NeerajMalhotra-QB committed:
Merge pull request #57 from NeerajMalhotra-QB/validate

change to consistent interface

NeerajMalhotra-QB reviewed (dismissed):
LGTM!!! 
1. merged meta-tags branch into develop
2. merged validate changes into develop. 
3. All UTs are passing

kasperjanehag reviewed (dismissed):
LGTM!

jaskaransinghsidana committed:
Changes to remove config yaml and introduce environment variables for parameterized runs

jaskaransinghsidana committed:
Changes to remove config yaml and introduce environment variables for parameterized runs

NeerajMalhotra-QB committed:
cleaning api/pyspark

NeerajMalhotra-QB committed:
backend and tests

NeerajMalhotra-QB committed:
Merge pull request #59 from NeerajMalhotra-QB/refactor

removing dead code and refactoring as needed

NeerajMalhotra-QB committed:
adding setter on errors accessors for pyspark

NeerajMalhotra-QB committed:
Merge pull request #60 from NeerajMalhotra-QB/getter_for_error

adding setter for errors object in pyspark accessor

NeerajMalhotra-QB committed:
reformatting error dict

NeerajMalhotra-QB committed:
Merge pull request #61 from NeerajMalhotra-QB/pretty_errs

reformatting error dict

jaskaransinghsidana committed:
Changes to rename the config object and call only in utils.py

jaskaransinghsidana committed:
Merge remote-tracking branch 'origin/feature_kill_switch' into feature_kill_switch

# Conflicts:
#	pandera/api/pyspark/container.py
#	pandera/api/pyspark/model.py
#	pandera/backends/pyspark/base.py
#	pandera/backends/pyspark/utils.py
#	tests/pyspark/conftest.py
#	tests/pyspark/test_pyspark_check.py
#	tests/pyspark/test_pyspark_error.py

jaskaransinghsidana committed:
Fixing merge conflict issue

jaskaransinghsidana committed:
Updating the test cases to support new checks types

jaskaransinghsidana committed:
Added individualized test for each configuration type.

jaskaransinghsidana committed:
Removing unnecessary prints

jaskaransinghsidana committed:
The changes include the following:
1) Fixed test case for validating the environment variable
2) Improved docstrings for test cases and few test cases asserts

jaskaransinghsidana committed:
Fix reference to with wrong key in test_pyspark_schema_data_checks

NeerajMalhotra-QB committed:
minor change

NeerajMalhotra-QB committed:
Merge pull request #58 from NeerajMalhotra-QB/feature_kill_switch

Changes to remove config yaml and introduce environment variables for parameterized runs

jaskaransinghsidana committed:
Added Support for docstring substitution method.

jaskaransinghsidana committed:
Removing an extra indent

jaskaransinghsidana committed:
Removing commented docstring substitution from __new__ method

NeerajMalhotra-QB committed:
remove union

NeerajMalhotra-QB committed:
cleaning

NeerajMalhotra-QB committed:
Merge pull request #64 from NeerajMalhotra-QB/ps_topics

Fixes to PySpark SQL Accessor

jaskaransinghsidana committed:
Feature to add metadata dictionary for pandas schema

jaskaransinghsidana committed:
Feature to add metadata dictionary for pandas schema

jaskaransinghsidana committed:
Added test to check the docstring substitution decorator

jaskaransinghsidana committed:
Added test to check the docstring substitution decorator

jaskaransinghsidana committed:
Merge pull request #63 from NeerajMalhotra-QB/feature_doc_string

Feature doc string substitution

jaskaransinghsidana committed:
Merge remote-tracking branch 'origin/feature_pandas_metadata' into feature_pandas_metadata

jaskaransinghsidana committed:
Changes to ensure only pandas run does not import pyspark dependencies

jaskaransinghsidana committed:
Fix of imports for pandas and pyspark for separation

jaskaransinghsidana committed:
Rename the function from pyspark to pandas

jaskaransinghsidana committed:
black lint and isort

jaskaransinghsidana committed:
black lint and isort

jaskaransinghsidana committed:
Merge remote-tracking branch 'origin/feature_doc_string' into feature_doc_string

jaskaransinghsidana committed:
Fixes of pyliny issue and suppression wherever necessary

NeerajMalhotra-QB committed:
Merge pull request #65 from NeerajMalhotra-QB/feature_pandas_metadata

Feature pandas metadata

NeerajMalhotra-QB committed:
Merge pull request #66 from NeerajMalhotra-QB/fix_linting_issue

Fix pylint failures

jaskaransinghsidana committed:
Fixes of mypy failures and redone black linting post changes.

NeerajMalhotra-QB committed:
Merge pull request #67 from NeerajMalhotra-QB/fix_linting_issue

Fixes of mypy failures and redone black linting post changes.

jaskaransinghsidana committed:
Added new test cases, removed redundant codes and black lint.

jaskaransinghsidana committed:
Fixed the doc strings, added functionality and test for custom checks

jaskaransinghsidana committed:
Fixed the doc strings, added functionality and test for custom checks

NeerajMalhotra-QB committed:
add rst for pyspark.sql

NeerajMalhotra-QB committed:
add rst for pyspark.sql

NeerajMalhotra-QB committed:
add rst for pyspark.sql

NeerajMalhotra-QB committed:
removing rst

NeerajMalhotra-QB committed:
removing rst

jaskaransinghsidana committed:
Renamed check name and Fixed pylint and mypy issues

jaskaransinghsidana committed:
Renamed check name and Fixed pylint and mypy issues

jaskaransinghsidana committed:
Merge pull request #68 from NeerajMalhotra-QB/fix_linting_issue

Added new test cases, removed redundant codes and black lint.

jaskaransinghsidana committed:
Merge remote-tracking branch 'origin/readme' into readme

jaskaransinghsidana committed:
Merge remote-tracking branch 'origin/feature_doc_string' into feature_doc_string

jaskaransinghsidana committed:
Merge remote-tracking branch 'origin/readme' into readme

jaskaransinghsidana committed:
Rename for environment variable key name

jaskaransinghsidana committed:
removing rst

jaskaransinghsidana committed:
Black lint

NeerajMalhotra-QB committed:
Merge pull request #69 from NeerajMalhotra-QB/feature_doc_string

Feature doc string

jaskaransinghsidana committed:
Removed daytime interval type

jaskaransinghsidana committed:
Merge pull request #70 from NeerajMalhotra-QB/removed_daytime_interval

Removed daytime interval type
